### PR TITLE
Redis connector support, based on new CloudFoundry service bindings

### PR DESCRIPTION
**Pull request by bart-vmware**
## Description

Builds on top of the CloudFoundry service bindings, providing IOptions support and health checks.

Additionally, this PR improves custom connection string builders (see commit message for details) and introduces shims. Shims provide a statically-typed API for reflection-based access to assemblies loaded at runtime and are intended to replace type locators. Because shims are non-static and take a set of input assemblies, we can add unit tests for both Pomelo/Oracle MySQL assemblies, which can be run in parallel.

For example, the code below uses shims to load types and access their properties:
```c#
var connectionMultiplexerShim = new ConnectionMultiplexerInterfaceShim(
    stackExchangeRedisPackageResolver, connectionMultiplexerInstance);

var redisCacheOptionsShim = RedisCacheOptionsShim.CreateInstance(microsoftRedisPackageResolver);
redisCacheOptionsShim.InstanceName = connectionMultiplexerShim.ClientName;
```
To compare, if we had a direct reference, the equivalent code would be:
```c#
IConnectionMultiplexer connectionMultiplexer = (ConnectionMultiplexer)connectionMultiplexerInstance;
var redisCacheOptions = new RedisCacheOptions();
redisCacheOptions.InstanceName = connectionMultiplexer.ClientName;
```

Once approved, I intend to use this pattern in the other connectors and add test coverage.

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [x] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [x] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

**Comment by bart-vmware** [comment]
Pushed more commits to:
- Change package/namespace from Steeltoe.Connector to Steeltoe.Connectors
- Rename ConnectionProvider to Connector, ConnectionFactory to ConnectorFactory

File: src/Connectors/src/Connectors/MongoDb/MongoDbConnectionStringBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+using Steeltoe.Common;
+
+namespace Steeltoe.Connectors.MongoDb;
+
+/// <summary>
+/// Provides a connection string builder for MongoDB URLs. The format is defined at: https://www.mongodb.com/docs/manual/reference/connection-string/.
+/// </summary>
+internal sealed class MongoDbConnectionStringBuilder : IConnectionStringBuilder
+{
+    private readonly Dictionary<string, string> _settings = new(StringComparer.OrdinalIgnoreCase);
```
Comment (TimHess) [inline src/Connectors/src/Connectors/MongoDb/MongoDbConnectionStringBuilder.cs:21]: I was going to note that it seems as though we might get into a case sensitivity issue later in this file with the unknown keywords, but it looks like you took care of that in the initialization already, nice!

**Review by TimHess — APPROVED** [review]
LGTM

**Pull request merged**